aws/sdk/examples/cognitoidentity/src/bin/list-identity-pools.rs

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use aws_types::region::{self, ProvideRegion};
+use cognitoidentity::{Client, Config, Error, Region};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+struct Opt {
+    /// The default region
+    #[structopt(short, long)]
+    default_region: Option<String>,
+
+    /// Whether to display additional information
+    #[structopt(short, long)]
+    verbose: bool,
+}
+
+/// Lists your Amazon Cognito identities
+/// # Arguments
+///
+/// * `[-d DEFAULT-REGION]` - The region containing the buckets.
+///   If not supplied, uses the value of the **AWS_DEFAULT_REGION** environment variable.
+///   If the environment variable is not set, defaults to **us-west-2**.
+/// * `[-g]` - Whether to display buckets in all regions.
+/// * `[-v]` - Whether to display additional information.
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    tracing_subscriber::fmt::init();
+
+    let Opt {
+        default_region,
+        verbose,
+    } = Opt::from_args();
+
+    let region = region::ChainProvider::first_try(default_region.map(Region::new))
```

Comment:
you don't actually need to call `.region()` anymore, you can pass in the provider.